include/cache.php

```diff
@@ -84,9 +84,14 @@ public static function get($key) {
 		$memcache = self::memcache();
 		if (is_object($memcache)) {
 			// We fetch with the hostname as key to avoid problems with other applications
-			$value = $memcache->get(get_app()->get_hostname().":".$key);
-			if (!is_bool($value)) {
-				return unserialize($value);
+			$cached = $memcache->get(get_app()->get_hostname().":".$key);
+			$value = @unserialize($cached);
+
+			// Only return a value if the serialized value is valid.
+			// We also check if the db entry is a serialized
+			// boolean 'false' value (which we want to return).
+			if ($cached === 'b:0;' || $value !== false) {
+				return $value;
```

Comment:
I really would suggest some error handling stuff. I hadn't worked with it either, but I guess there will be enough samples out there.